File: python/individual_sanctions_list_test.py
from onboarding_buddy_client.api.sanctions_api import SanctionsApi
from onboarding_buddy_client.models.individual_sanctions_check_request_m import IndividualSanctionsCheckRequestM
from onboarding_buddy_client import Configuration, ApiClient
from dotenv import load_dotenv
import os

load_dotenv()

# Configure the client with authentication headers
config = Configuration(
    api_key={
        "AppKey": os.getenv("OB_APP_KEY"),
        "ApiKey": os.getenv("OB_API_KEY"),
        "ApiSecret": os.getenv("OB_API_SECRET")
    }
)

# Initialize the API client
api_client = ApiClient(configuration=config)
sanctions_api = SanctionsApi(api_client)

# Submit the email validation request
try:
    # Create an individual sanction request
    request = IndividualSanctionsCheckRequestM(
        first_name="YEVGENIY",
        last_name="PRIGOZHIN",
        birth_year="1961"
    )

    response = sanctions_api.individual(individual_sanctions_check_request_m=request)
    
    print(f"Matched: {response.matched}")
    print(f"Full Name: {response.results[0].full_name}")
    print(f"First Name: {response.results[0].first_name}")
    print(f"Last Name: {response.results[0].last_name}")
    print(f"Program: {response.results[0].program}")
    print(f"Additional Information: {response.results[0].additional_info}")
    print(f"Date Of Birth: {response.results[0].date_of_birth}")
    print(f"Birth Year: {response.results[0].birth_year}")
    print(f"Nationality: {response.results[0].nationality}")
    print(f"Gender: {response.results[0].gender}")
    print(f"Linked To: {response.results[0].linked_to}")
    print(f"Secondary Sanctions Risk: {response.results[0].secondary_sanctions_risk}")
    
except Exception as e:
    print(f"Error: {e}")

File: typescript/src/individual_sanctions_list_test.ts
import OnboardingBuddyClient from '@onboardingbuddy/onboardingbuddy-client'
import * as dotenv from 'dotenv'
dotenv.config()

const obAppKey = process.env.OB_APP_KEY || '';
const obApiKey = process.env.OB_API_KEY || '';
const obApiSecret = process.env.OB_API_SECRET || '';

const client = new OnboardingBuddyClient(obAppKey, obApiKey, obApiSecret);

const individualSanctions = client.checkIndividualSanctions({
  firstName: 'YEVGENIY',
  lastName: 'PRIGOZHIN',
  birthYear: '1961',
});

individualSanctions
    .then(response => 
        { 
            console.log(`Matched: ${response.data.matched}`)

            if (response.data.results)
            {
                console.log(`Full Name: ${response.data.results[0].fullName}`)
                console.log(`First Name: ${response.data.results[0].firstName}`)
                console.log(`Last Name: ${response.data.results[0].lastName}`)
                console.log(`Program: ${response.data.results[0].program}`)
                console.log(`Additional Information: ${response.data.results[0].additionalInfo}`)
                console.log(`Date Of Birth: ${response.data.results[0].dateOfBirth}`)
                console.log(`Birth Year: ${response.data.results[0].birthYear}`)
                console.log(`Nationality: ${response.data.results[0].nationality}`)
                console.log(`Gender: ${response.data.results[0].gender}`)
                console.log(`Linked To: ${response.data.results[0].linkedTo}`)
                console.log(`Secondary Sanctions Risk: ${response.data.results[0].secondarySanctionsRisk}`)
            }
        })
    .catch(error=> console.log(error.response.data));
